main.py

```diff
@@ -0,0 +1,42 @@
+import os
+
+
+def get_contents_of_one_file(file_content):
+    file_values = []
+    for row in file_content.split("\n")[1:-1]:
+        file_values.append(row.split(","))
+
+    return file_values
+
+
+def read_files():
+    file_names = os.listdir("weatherfiles")
+
+    # file_reader = open("main.py", "r")
+    # file_content = file_reader.read()
+    # file_reader.close()
+
+    file_values = []
+
+    for file_name in file_names:
+        with open(f"weatherfiles/{file_name}", "r") as file_reader:
+            file_content = file_reader.read()
+
+            file_values += get_contents_of_one_file(file_content)
+            file_reader.close()
+
+    return file_values
+
+
+file_values = read_files()
+
+temperatures = {}
+for file_value in file_values:
+    if file_value and file_value[1]:
+        temperatures[file_value[0]] = float(file_value[1])
+
+min_temp_year = min(temperatures, key=temperatures.get)
+max_temp_year = max(temperatures, key=temperatures.get)

```

Comment:
How is this actually working. what is `key` arg here?

Reply:
key= lgany se pehly programe run ho raha tha lekin wo ye kar raha tha k wo file me gya or usny dekha temperature ko pehly usny first line me dekha temp 20 he next line me tmep 21 he to progrmame wahi pe ruk gya agy nai ja raha tha key= lgany k baad wo sari files me  ja k har 1 line ko read kar raha he or jo min() function lgaya he wo minimum number find karny k lya lgaya he ab wo minimum ko find kary ga chahy hmary pass 100000 file hi q na ho sab ja k sab se kam ko nikal k laye ga


next line me max() or key same yahi kaam he lekin yaha pe wo maximum ko nikal k laye ga

Reply:
good, now do it without `min` or `max` function

Reply:
ok